### PR TITLE
Introduce CRIU Support SystemRestoreException

### PR DESCRIPTION
**Pull request by tajila**
Introduce CRIU Support SystemRestoreException

Depends on https://github.com/eclipse-openj9/openj9/pull/16251

Signed-off-by: Tobi Ajila <atobia@ca.ibm.com>

### CODE REVIEW

**Comment by tajila** [comment]
@tjwatson please review

**Comment by tajila** [comment]
@tjwatson The Openj9 changes have been merged

File: dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/criu/CheckpointFailedException.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public enum Type {
         UNKNOWN_CHECKPOINT(75, false),
 
         JVM_RESTORE_FAILED(76, true),
+        SYSTEM_RESTORE_FAILED(79, true),
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/criu/CheckpointFailedException.java]: I would put this before `JVM_RESTORE_FAILED(76, true),` and make `SYSTEM_RESTORE_FAILED`  use 76 return code.  Shift down all the rest:

```java
        SYSTEM_RESTORE_FAILED(76, true),
        JVM_RESTORE_FAILED(77, true),
        LIBERTY_RESTORE_FAILED(78, true),
        UNKNOWN_RESTORE(79, true);
```

Maybe best to move all the restore fails to start at error code 80 to give us a gap between the checkpoint failures (which start at 70) and the restore failures:

```java
        SYSTEM_RESTORE_FAILED(80, true),
        JVM_RESTORE_FAILED(81, true),
        LIBERTY_RESTORE_FAILED(82, true),
        UNKNOWN_RESTORE(83, true);
```
I'm not sure how many tests that could cause to fail, but we can adjust them.

File: dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/openj9/ExecuteCRIU_OpenJ9.java [hunk 2]
```diff
@@ -59,8 +61,10 @@ public void dump(Runnable prepare, Runnable restore, File imageDir, String logFi
             throw new CheckpointFailedException(Type.JVM_CHECKPOINT_FAILED, e.getMessage(), e);
         } catch (SystemCheckpointException e) {
             throw new CheckpointFailedException(Type.SYSTEM_CHECKPOINT_FAILED, e.getMessage(), e);
-        } catch (RestoreException e) {
+        } catch (JVMRestoreException e) {
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/openj9/ExecuteCRIU_OpenJ9.java]: We may need to do this in the catch for `RuntimeException e` below such that we can stage this in to work on the current EA OpenJ9 build as well as the new one.  Something like:

```java
} catch (RuntimeException e) {
    String exceptionName = e.getClass().getSimpleName();
    if ("JVMRestoreException".equals(exceptionName) || "RestoreException".equals(exceptionName)) {
        throw new CheckpointFailedException(Type.JVM_RESTORE_FAILED, e.getMessage(), e);
    }
    if ("SystemRestoreException".equals(exceptionName)) {
        throw new CheckpointFailedException(Type.SYSTEM_RESTORE_FAILED, e.getMessage(), e);
    }
    throw new CheckpointFailedException(checkpointImpl.getUnknownType(), e.getMessage(), e);
}
```
Then once we have an EA with the API change and we have moved all the infrastructure to the new Open J9 EA build then we can move back to actual catch blocks with the new exceptions.

**Comment by tjwatson** [comment]
#build
#libby

**Review by tjwatson — APPROVED** [review]

**Comment by mbroz2** [comment]
#build

**Comment by tjwatson** [comment]
PR #23575 built this with:

https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_o6p6UHJnEe2SBN8svFwMrQ

This build has one error that is not related to this PR.  Going to merge this.

**Pull request merged**